pkg/kubelet/server/server.go

```diff
@@ -431,6 +433,21 @@ func (s *Server) syncLoopHealthCheck(req *http.Request) error {
 	return nil
 }
 
+// Checks if any of the kubelet's reflectors have reported any errors
+func (s *Server) reflectorHealthCheck(req *http.Request) error {
+	healthy, reflectorErrors := s.host.ReflectorsHealthy()
+	if healthy {
+		return nil
+	}
+
+	reflectorErrorStrings := make([]string, 0, len(reflectorErrors))
+	for _, reflectorError := range reflectorErrors {
+		reflectorErrorStrings = append(reflectorErrorStrings, reflectorError.Error())
+	}
+
+	return fmt.Errorf("Reflector errors: %s", strings.Join(reflectorErrorStrings, ", "))
```

Comment:
Maybe just use `errors.NewAggregate(...)` here? 
https://github.com/kubernetes/kubernetes/blob/f0a061e361bccd7080e6adb0674b0c02582608e8/staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go#L39
Then you could just simplify the whole method to this, since `NewAggregate` handles the empty list:
```go
_, reflectorErrors := s.host.ReflectorsHealthy()
return errors.NewAggregate(reflectorErrors)
```